### PR TITLE
Add leaving channel

### PR DESCRIPTION
**Pull request by adasiewiczr**
### Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [x] Changelog updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [x] Reviewers added

### CODE REVIEW

File: stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/controller/QueryChannelsControllerImpl.kt [hunk 1]
```diff
@@ -296,6 +296,20 @@ internal class QueryChannelsControllerImpl(
...
+        // Remove from queryChannelsSpec
+        if (queryChannelsSpec.cids.contains(cid)) {
+            queryChannelsSpec.cids = queryChannelsSpec.cids - cid
+            domainImpl.repos.queryChannels.insert(queryChannelsSpec)
+            // Remove from channel repository
+            domainImpl.repos.removeChannel(cid)
+
+            _channels.value = _channels.value.toMutableMap().apply {
+                remove(cid)
+            }
```
Comment (zsmb13) [inline stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/controller/QueryChannelsControllerImpl.kt:309]:
```suggestion
            _channels.value = _channels.value.minus(cid)
```

File: stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/viewmodel/channels/ChannelsViewModel.kt [hunk 2]
```diff
@@ -82,6 +82,10 @@ public class ChannelsViewModel(
         }.exhaustive
     }
 
+    public fun leaveChannel(channel: Channel) {
+        chatDomain.useCases.leaveChannel(channel.cid).enqueue()
+    }
```
Comment (zsmb13) [inline stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/viewmodel/channels/ChannelsViewModel.kt:87]: This is lacking some error handling, but not sure if we have any specific behaviour we'd want here on an error

**Review by zsmb13 — APPROVED** [review]
Looks good to me in general 👍

File: stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/viewmodel/channels/ChannelsViewModel.kt (hunk 2 above)
Comment (adasiewiczr) [reply stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/viewmodel/channels/ChannelsViewModel.kt:87]: For now, channel won't be removed from the list without any error. We also don't handle error in hiding channel method so I would leave it for now, but definitely, we need to think how to handle such cases

**Review by zsmb13 — APPROVED** [review]

**Pull request merged**